modules/eks/cluster/README.md

```diff
@@ -73,40 +153,86 @@ components:
         github_actions_allowed_repos:
           - acme/infra
 
-        # We use karpenter to provision nodes
-        # See below for using node_groups
-        managed_node_groups_enabled: false
-        node_groups: {}
-
-        # EKS IAM Authentication settings
-        # By default, you can authenticate to EKS cluster only by assuming the role that created the cluster.
-        # After the Auth Config Map is applied, the other IAM roles in
-        # `primary_iam_roles`, `delegated_iam_roles`, and `sso_iam_roles` will be able to authenticate.
-        apply_config_map_aws_auth: true
+        # We recommend, at a minimum, deploying 1 managed node group,
+        # with the same number of instances as availability zones (typically 3).
+        managed_node_groups_enabled: true
+        node_groups: # for most attributes, setting null here means use setting from node_group_defaults
+          main:
+            # availability_zones = null will create one autoscaling group
+            # in every private subnet in the VPC
+            availability_zones: null
+
+            # Tune the desired and minimum group size according to your baseload requirements.
+            # We recommend no autoscaling for the main node group, so it will
+            # stay at the specified desired group size, with additional
+            # capacity provided by Karpenter. Nevertheless, we recommend
+            # deploying enough capacity in the node group to handle your
+            # baseload requirements, and in production, we recommend you
+            # have a large enough node group to handle 3/2 (1.5) times your
+            # baseload requirements, to handle the loss of a single AZ.
+            desired_group_size: 3 # number of instances to start with, should be >= number of AZs
+            min_group_size: 3 # must be  >= number of AZs
+            max_group_size: 3
```

Comment:
This have been a point confusion for multiple end users. If you have 3 AZ's, setting this to 5 or to 3 is effectively rendered the same and has no effect. This is due to [this line in the node group module](https://github.com/cloudposse/terraform-aws-components/blob/main/modules/eks/cluster/modules/node_group_by_region/main.tf#L13-L15)

Really, we should change this variable to be more clear, like `desired_per_az` and `min_per_az`. 

```suggestion
            desired_group_size: 3 # number of instances to start with, should be a multiple of the # of AZs you have
            min_group_size: 3 # must be a multiple of the # of AZs you have
            max_group_size: 3 # must be a multiple of the # of AZs you have
```

Reply:
@dudymas You are mistaken. 

When you use 3 AZs:
- If you set `max_group_size` to 3, you get 3 node groups/ASGs (one per AZ), each with a `max_size` of 1
- If you set `max_group_size` to 5, you get 3 node groups/ASGs (one per AZ), 1 of which has a `max_size` of 1, while the other 2 have `max_size` of 2

Reply:
Desk-checked this again. You're right, and I missed the way index was getting added. As for the behavior in the past, it must have been a different module I'm getting this confused with.